enhancements/windows-containers/csi-proxy.md

```diff
@@ -0,0 +1,160 @@
+will ensure that csi-proxy is running on the Windows nodes it configures. This will be done through the normal
+workflow for adding a service to the expected service list.
+
+The csi-proxy binary will be baked into the WMCO image. WMCO will copy the binary to each node at configuration
+time. A new item will be added to the windows-services ConfigMap, instructing the windows-instance-config-daemon
+to ensure that csi-proxy is always running as a Windows service on each node.
+
+With the nodes in this state, users are free to follow upstream documentation to deploy the CSI node drivers
+of their choice on the Windows nodes in the cluster.
+
+No support from Red Hat will be given for issues occurring with the upstream drivers, as they are not being distributed
+as part of OpenShift.
+
+
+### Workflow Description
+
+1. The cluster administrator will identify what CSI storage drivers the cluster requires for its workloads.
+2. The cluster admin can follow upstream documentation to deploy the CSI node drivers on its Windows nodes.
+
+### API Extensions
+
+N/A
+
+### Risks and Mitigations
+
+If there are any security issues present in the CSI driver images deployed by the user, they will not be fixed
+by upgrading OCP. It is up to the user to stay informed and update their images as necessary.
+
+### Drawbacks
+
+The main drawback to this approach is it puts a lot of work on the cluster administrator. As they will be required
+to deploy and maintain the Windows CSI DaemonSet themselves, and will not be given support for any issues that may
+come up, which do not specifically deal with the csi-proxy Windows service. This may cause an increased load
+on both support and developers, as support issues coming in will need to be properly filtered out based on the root
+cause of the problem.
+
+## Design Details
+
+### Test Plan
+
+WMCO already has an extensive end to end test suite. To facilitate testing the additions described in this enhancement,
+the e2e tests will be augmented to deploy a CSI node driver DaemonSet across Windows nodes, just as a user would. The
+e2e tests would then validate that a Windows pod is able to mount a volume backed by the driver.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+This feature is targeted for WMCO 9.0.0 in OpenShift 4.14.
+
+#### Removing a deprecated feature
+
+No action required.
+
+### Upgrade / Downgrade Strategy
+
+[When upgrading cluster versions, in-tree driver Volumes will automatically be updated to use the appropriate CSI node
+drivers by the cluster](https://docs.openshift.com/container-platform/4.12/storage/container_storage_interface/persistent-storage-csi-migration.html).
+
+If users are making use of the Linux CSI node drivers on the cluster, and not deploying all parts of storage
+themselves, some extra care must be had when upgrading a cluster.
```

Comment:
What is this extra care?

Reply:
The extra care is what is written in the rest of this section

Reply:
Then please add `as described below`.